pkg/tls/cred.go

```diff
@@ -90,7 +91,17 @@ func (crt *Crt) Verify(roots *x509.CertPool, name string) error {
 	}
 	vo := x509.VerifyOptions{Roots: roots, Intermediates: i, DNSName: name}
 	_, err := crt.Certificate.Verify(vo)
-	return err
+
+	//If error due to expiry
+	switch err.(type) {
+	case x509.CertificateInvalidError:
+		expTime := crt.Certificate.NotAfter
```

Comment:
Two things that stand out. I think here you need to test whether the reason in the `CertificateInvalidError` is `Expired` as I already mentioned. This cert can be invalid for other reasons. Also, its good to include the NotBefore time as well as it could be that its just not valid yet.